utils/zerovec/src/ule/mod.rs

```diff
@@ -66,11 +83,19 @@ where
     ///
```

Comment:
Please add a default impl for `from_byte_slice_unchecked` as well.  For ULE, you can use `mem::size_of` to figure out how long to make the resulting slice.  For VarULE, I think you can just cast it, right?

Reply:
You can't cast it for VarULE because there's no way to generically do the inverse operation of `size_of_val()` ("I have X bytes, tell me what the pointer metadata should be"). For example, consider what would happen with stuff like `struct ComplexULE(u8, char::ULE, [u32::ULE])`, calculating the pointer metadata for that is very specific to that type.

Now _fortunately_ the only kind of DST we can theoretically support are slice DSTs and compound slice DSTs, but just to give an example of why the inverse of `size_of_val()` is untractable: how woudl you solve that for trait objects? At least for compound slice-based DSTs rustc could theoretically add APIs to `std::ptr::metadata` that let you do this.